lib/Dialect/CGGI/Conversions/CGGIToTfheRust/CGGIToTfheRust.cpp

```diff
@@ -175,9 +175,12 @@ struct ConvertLut3Op : public OpConversionPattern<cggi::Lut3Op> {
         serverKey, adaptor.getC(), b.getIndexAttr(2));
     auto shiftedB = b.create<tfhe_rust::ScalarLeftShiftOp>(
         serverKey, adaptor.getB(), b.getIndexAttr(1));
-    auto summedBC = b.create<tfhe_rust::AddOp>(serverKey, shiftedC, shiftedB);
-    auto summedABC =
-        b.create<tfhe_rust::AddOp>(serverKey, summedBC, adaptor.getA());
+    auto outputType =
+        getTypeConverter()->convertType(shiftedB.getResult().getType());
```

Comment:
interesting, just curious - why is this not the same type as `adaptor.getB()` (even though the all types match constraint is removed, should this whole sequence of ops be the same type as adaptor.getC() / getB() / getA())?